packages/ui/src/components/Visualization/Canvas/Canvas.tsx

```diff
@@ -92,7 +92,7 @@ export const Canvas: FunctionComponent<PropsWithChildren<CanvasProps>> = ({ enti
 
     controller.fromModel(model, true);
     setInitialized(true);
-  }, [controller, entities, visibleFlows]);
+  }, [activeLayout, controller, entities, visibleFlows]);
```

Comment:
Hi @ibek, thanks for catching this missing dependency, but in this case, `activeLayout` is purposely omitted.

If you're curious, continue reading 😃 
When the layout is changed, the `Canvas.tsx` is rendered, triggering this `useEffect` again, and this is not needed since the underlying canvas knows when the `activeLayout` changes and automatically updates without needing to recalculate the entities from the outside.

```suggestion
  }, [controller, entities, visibleFlows]);
```

Reply:
Hi, makes sense, thank you